### PR TITLE
baremetal: Convert templates to steps

### PR DESCRIPTION
**Pull request by rna-afk**
Converting baremetal UPI tests from templates to steps.

### CODE REVIEW

**Comment by patrickdillon** [comment]
generated config is failing with:
```
ERROR: This check enforces that Prow Job configuration YAML files are generated
ERROR: correctly. We have automation in place that generates these configs and
ERROR: new changes to these job configurations should occur from a re-generation.
ERROR: Run the following command to re-generate the Prow jobs:
ERROR: $ make jobs
```

I wonder if that would unstick the error we're seeing in pj-rehearse?

**Comment by rna-afk** [comment]
Not able to run jobs.
```
~/go/github.com/openshift/release$ sudo make jobs
Place your finger on the fingerprint reader
docker pull registry.ci.openshift.org/ci/ci-operator-checkconfig:latest
latest: Pulling from ci/ci-operator-checkconfig
Digest: sha256:3d979ed537ccf053a4c942b67a4f0b49108499300c00f61718ce23e1e03de2ab
Status: Image is up to date for registry.ci.openshift.org/ci/ci-operator-checkconfig:latest
registry.ci.openshift.org/ci/ci-operator-checkconfig:latest
docker run --rm -v "/home/anarayan/go/github.com/openshift/release/ci-operator/config:/ci-operator/config:z" -v "/home/anarayan/go/github.com/openshift/release/ci-operator/step-registry:/ci-operator/step-registry:z" registry.ci.openshift.org/ci/ci-operator-checkconfig:latest --config-dir /ci-operator/config --registry /ci-operator/step-registry
time="2022-05-26T13:38:14Z" level=error msg="Failed to load CI Operator configuration" error="invalid ci-operator config: invalid configuration: releases[stable-4.9]: the release name is not valid: must not contain '.'" source-file=/ci-operator/config/openshift/assisted-service/openshift-assisted-service-release-ocm-2.5.yaml
time="2022-05-26T13:38:14Z" level=error msg="Failed to load CI Operator configuration" error="invalid ci-operator config: invalid configuration: importing the release stream is required to ensure the promoted images to the namespace ocp can be integrated properly. Although it can be achieved by tag_specification or releases[\"latest\"], adding an e2e test is strongly suggested" source-file=/ci-operator/config/openshift/custom-metrics-autoscaler-operator/openshift-custom-metrics-autoscaler-operator-main.yaml
time="2022-05-26T13:38:14Z" level=error msg="Failed to load CI Operator configuration" error="invalid ci-operator config: invalid configuration: importing the release stream is required to ensure the promoted images to the namespace ocp can be integrated properly. Although it can be achieved by tag_specification or releases[\"latest\"], adding an e2e test is strongly suggested" source-file=/ci-operator/config/openshift/kedacore-keda/openshift-kedacore-keda-main.yaml
time="2022-05-26T13:38:15Z" level=error error="error reading configuration files: [invalid ci-operator config: invalid configuration: releases[stable-4.9]: the release name is not valid: must not contain '.', invalid ci-operator config: invalid configuration: importing the release stream is required to ensure the promoted images to the namespace ocp can be integrated properly. Although it can be achieved by tag_specification or releases[\"latest\"], adding an e2e test is strongly suggested]"
time="2022-05-26T13:38:15Z" level=fatal msg="error validating configuration files"

```

File: ci-operator/step-registry/upi/deprovision/metal/upi-deprovision-metal-commands.sh [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+for artifact in must-gather audit-logs ; do
+tar -czC ${ARTIFACT_DIR}/${artifact} -f ${ARTIFACT_DIR}/${artifact}.tar.gz . &&
+rm -rf ${ARTIFACT_DIR:?}/${artifact}
+done
```
Comment (patrickdillon) [inline ci-operator/step-registry/upi/deprovision/metal/upi-deprovision-metal-commands.sh:124]: I think almost all of this stuff is already in gather-extra: https://steps.ci.openshift.org/reference/gather-extra#line205

In which case, let's remove it and make sure gather extra is run.

**Comment by patrickdillon** [comment]
> Not able to run jobs.
> 
> ```
> ~/go/github.com/openshift/release$ sudo make jobs
> Place your finger on the fingerprint reader
> docker pull registry.ci.openshift.org/ci/ci-operator-checkconfig:latest
> latest: Pulling from ci/ci-operator-checkconfig
> Digest: sha256:3d979ed537ccf053a4c942b67a4f0b49108499300c00f61718ce23e1e03de2ab
> Status: Image is up to date for registry.ci.openshift.org/ci/ci-operator-checkconfig:latest
> registry.ci.openshift.org/ci/ci-operator-checkconfig:latest
> docker run --rm -v "/home/anarayan/go/github.com/openshift/release/ci-operator/config:/ci-operator/config:z" -v "/home/anarayan/go/github.com/openshift/release/ci-operator/step-registry:/ci-operator/step-registry:z" registry.ci.openshift.org/ci/ci-operator-checkconfig:latest --config-dir /ci-operator/config --registry /ci-operator/step-registry
> time="2022-05-26T13:38:14Z" level=error msg="Failed to load CI Operator configuration" error="invalid ci-operator config: invalid configuration: releases[stable-4.9]: the release name is not valid: must not contain '.'" source-file=/ci-operator/config/openshift/assisted-service/openshift-assisted-service-release-ocm-2.5.yaml
> time="2022-05-26T13:38:14Z" level=error msg="Failed to load CI Operator configuration" error="invalid ci-operator config: invalid configuration: importing the release stream is required to ensure the promoted images to the namespace ocp can be integrated properly. Although it can be achieved by tag_specification or releases[\"latest\"], adding an e2e test is strongly suggested" source-file=/ci-operator/config/openshift/custom-metrics-autoscaler-operator/openshift-custom-metrics-autoscaler-operator-main.yaml
> time="2022-05-26T13:38:14Z" level=error msg="Failed to load CI Operator configuration" error="invalid ci-operator config: invalid configuration: importing the release stream is required to ensure the promoted images to the namespace ocp can be integrated properly. Although it can be achieved by tag_specification or releases[\"latest\"], adding an e2e test is strongly suggested" source-file=/ci-operator/config/openshift/kedacore-keda/openshift-kedacore-keda-main.yaml
> time="2022-05-26T13:38:15Z" level=error error="error reading configuration files: [invalid ci-operator config: invalid configuration: releases[stable-4.9]: the release name is not valid: must not contain '.', invalid ci-operator config: invalid configuration: importing the release stream is required to ensure the promoted images to the namespace ocp can be integrated properly. Although it can be achieved by tag_specification or releases[\"latest\"], adding an e2e test is strongly suggested]"
> time="2022-05-26T13:38:15Z" level=fatal msg="error validating configuration files"
> ```

Your comment from standup was accurate: rebasing fixed this issue for me.

**Comment by patrickdillon** [comment]
Hahaha, CI ran the survey:
```
Installing from release registry.build04.ci.openshift.org/ci-op-b5sxmklj/release@sha256:55071eea67cac745fd2717dde2b092bb9dfe66594d43df03f6c5aa5be5a247b1
[0G[2K[1;92m? [0m[1;99mPlatform[0m  [36m[Use arrows to move, type to filter, ? for more help][0m
[1;36m> alibabacloud[0m
[39m  aws[0m
[39m  azure[0m
[39m  gcp[0m
[39m  ibmcloud[0m
[39m  nutanix[0m
[39m  openstack[0m
[?25l[?25hlevel=fatal msg=failed to fetch Master Machines: failed to fetch dependency of "Master Machines": failed to fetch dependency of "Cluster ID": failed to fetch dependency of "Install Config": failed to fetch dependency of "Base Domain": failed to generate asset "Platform": EOF
```
Never seen that before...

**Comment by rna-afk** [comment]
Closing since we decided to strip all the metal jobs in CI 
https://coreos.slack.com/archives/C68TNFWA2/p1656341285550539

**Pull request closed without merging**